onyo/lib/items.py

```diff
@@ -0,0 +1,163 @@
+        if self.repo and self._path and self._path.name == self.repo.ASSET_DIR_FILE_NAME:
+            return self._path.parent
+        return self._path
+
+    def get_path_relative(self):
+        """Initializer for the 'onyo.path.relative' pseudo-key."""
+        return self['onyo.path.absolute'].relative_to(self.repo.git.root) if self.repo else None
+
+    def get_path_parent(self):
+        """Initializer for the 'onyo.path.parent' pseudo-key."""
+        return self['onyo.path.absolute'].parent.relative_to(self.repo.git.root) if self.repo else None
+
+    def get_path_file(self):
+        """Initializer for the 'onyo.path.file' pseudo-key."""
+        return self._path.relative_to(self.repo.git.root) if self.repo and self['onyo.is.asset'] else None
+
+    def is_asset(self) -> bool | None:
+        """Initializer for the 'onyo.is.asset' pseudo-key."""
+        if not self.repo or not self._path:
+            return None
+        return self.repo.is_asset_path(self._path)
+
+    def is_directory(self) -> bool | None:
+        """Initializer for the 'onyo.is.directory' pseudo-key."""
+        if not self.repo or not self._path:
+            return None
+        return self.repo.is_inventory_dir(self._path)
+
+    def is_template(self) -> bool | None:
+        """Initializer for the 'onyo.is.template' pseudo-key."""
+        if not self.repo or not self._path:
+            return None
+        return self.repo.git.root / self.repo.TEMPLATE_DIR in self._path.parents
+
+    def is_empty(self) -> bool | None:
+        """Initializer for the 'onyo.is.empty' pseudo-key."""
+        return None  # TODO: Unclear what exactly this needs to consider. Probably child items? Or just assets?
+
+# TODO/Notes for next PR(s):
+# - Bug/Missing feature: pseudo-keys that are supposed to be settable by commands, are not yet
+#   ensured to return bool/Path objects that the codebase acts upon when their values are coming in from
+#   CLI or (template-)files, since everything is stringified now.
+# - We need plain files/directories represented for --yaml
+# - Templates can be (asset-) dirs  -> a dir is an item. If it has the dot yaml file, it's an asset dir
+# - values in templates maybe to be evaluated matching expression or even plugin calls  #714
+# - Path attribute caching at git/onyo layers (is_file, etc.):
+#   We actually know that everything we get from
+#   git ls-tree is in fact a file or symlink. And we can derive
+#   dirs from that path list (.anchor). That may be a lot faster.
+#   Implement cache dict at GitRepo level.
+# - Probably/Maybe: Stop passing dicts and Path objects around. All things relevant at higher level are Items, right?
+# - suck in DotNotationWrapper instead of deriving!? Probably not, because we have asset/template specs that should
+#   behave with dot notation, but can't or even must not have pseudo-keys.
+# - What about `__eq__` (see the horrible is_equal_dict helper)?
```

Comment:
You read my mind. This was also something I was hoping the new object would allow us to address.

Reply:
Yeah, just wanted to see to what extent DotNotationWrapper or even regular dicts keep a role first (where something like that helper may be needed). In any case: At least part if not all of that helper will be simplified and go into `DotNotationWrapper/Item.__eq__`.